cluster/storage-cluster.go

```diff
@@ -0,0 +1,250 @@
+			}
+			return
+		}
+		defer stmt.Close()
+
+		var tenantId int32
+		err = stmt.QueryRow(scName).Scan(&tenantId)
+		if err != nil {
+			ch <- AddStorageClusterResult{
+				Error: err,
+			}
+			return
+		}
+		// return result via channel
+		ch <- AddStorageClusterResult{
+			StorageCluster: &StorageCluster{
+				Id:   tenantId,
+				Name: scName,
+			},
+			Error: nil,
+		}
+
+	}()
+	return ch
+}
+
+// provisions the storage cluster supporting services that point to each node in the storage cluster
+func ProvisionServicesForStorageCluster(storageCluster *StorageCluster) chan error {
+	ch := make(chan error)
+	go func() {
+		defer close(ch)
+		if storageCluster == nil {
+			ch <- errors.New("Empty storage cluster received")
+			return
+		}
+		for i := 1; i <= MaxNumberHost; i++ {
+			CreateSCHostService(
+				fmt.Sprintf("%d", storageCluster.Id),
+				fmt.Sprintf("%d", i),
+				nil)
+		}
+	}()
+	return ch
+}
+
+// Selects from all the available storage clusters for one with space available.
+func SelectSCWithSpace(ctx *Context) chan *StorageClusterResult {
+	ch := make(chan *StorageClusterResult)
+	go func() {
+		defer close(ch)
+		var id int32
+		var name *string
+		// For now, let's select a storage cluster at random
+		query := `
+			SELECT 
+			       id, name 
+			FROM 
+			     m3.provisioning.storage_clusters 
+			OFFSET 
+				floor(random() * (SELECT COUNT(*) FROM m3.provisioning.storage_clusters)) LIMIT 1;`
+
+		err := ctx.Tx.QueryRow(query).Scan(&id, &name)
+		if err != nil {
+			ch <- &StorageClusterResult{Error: err}
+			return
+		}
+		ch <- &StorageClusterResult{
+			StorageCluster: &StorageCluster{
+				Id:   id,
+				Name: name,
+			},
+		}
+
+	}()
+	return ch
+}
+
+func GetListOfTenantsForSCluster(ctx *Context, sc *StorageCluster) chan []*StorageClusterTenant {
+	ch := make(chan []*StorageClusterTenant)
+	go func() {
+		defer close(ch)
+		if sc == nil {
+			return
+		}
+		query := `
+			SELECT 
+			       t1.tenant_id, t1.port, t1.service_name, t2.name, t2.short_name 
+			FROM 
+			     m3.provisioning.tenants_storage_clusters t1
+			LEFT JOIN m3.provisioning.tenants t2
+			ON t1.tenant_id = t2.id
+			WHERE storage_cluster_id=$1`
+		rows, err := ctx.Tx.Query(query, sc.Id)
+		if err != nil {
+			fmt.Println(err)
+			return
+		}
+		var tenants []*StorageClusterTenant
+		for rows.Next() {
+			var tenantId int32
+			var tenantName string
+			var tenantShortName string
+			var port int32
+			var serviceName string
+			err = rows.Scan(&tenantId, &port, &serviceName, &tenantName, &tenantShortName)
+			if err != nil {
+				fmt.Println(err)
+			}
+
+			tenants = append(tenants, &StorageClusterTenant{
+				Tenant: &Tenant{
+					Id:        tenantId,
+					Name:      tenantName,
+					ShortName: tenantShortName,
+				},
+				Port:             port,
+				ServiceName:      serviceName,
+				StorageClusterId: sc.Id})
+
+		}
+		ch <- tenants
+	}()
+	return ch
+}
+
+type StorageClusterTenant struct {
+	*Tenant
+	StorageClusterId int32
```

Comment:
Are we limited to 4 bytes IDs? - Shouldn't that be a string?

Reply:
ideally the storage cluster id is a sequential number (ie sc-1, sc-2, sc-3) because it's convenient to use for auto-naming the storage clusters, so I hope it's enough.